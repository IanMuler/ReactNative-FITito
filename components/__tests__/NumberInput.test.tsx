/* Testing:
 * - Renders correctly with default value
 * - Increments the value
 * - Decrements the value
 * - Changes the value through text input
 */

import React from "react";
import renderer, { act } from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import NumberInput from "../NumberInput";

describe("NumberInput", () => {
  it("renders correctly with default value", () => {
    const tree = renderer
      .create(<NumberInput value="10" onChangeText={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("increments the value", () => {
    const onChangeText = jest.fn();
    const { getByTestId } = render(
      <NumberInput value="10" onChangeText={onChangeText} />
    );

    const incrementButton = getByTestId("increment-button");
    fireEvent.press(incrementButton);
    expect(onChangeText).toHaveBeenCalledWith("11");
  });

  it("decrements the value", () => {
    const onChangeText = jest.fn();
    const { getByTestId } = render(
      <NumberInput value="10" onChangeText={onChangeText} />
    );

    const decrementButton = getByTestId("decrement-button");
    fireEvent.press(decrementButton);
    expect(onChangeText).toHaveBeenCalledWith("9");
  });

  it("changes the value through text input", () => {
    const onChangeText = jest.fn();
    const { getByTestId } = render(
      <NumberInput value="10" onChangeText={onChangeText} />
    );

    const input = getByTestId("number-input");
    fireEvent.changeText(input, "15");
    expect(onChangeText).toHaveBeenCalledWith("15");
  });
});
