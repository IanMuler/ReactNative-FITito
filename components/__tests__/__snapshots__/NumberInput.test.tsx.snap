// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly with default value 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#293048",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "paddingHorizontal": 5,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="decrement-button"
  >
    <Text />
  </View>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholderTextColor="#A5A5A5"
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
        "fontSize": 16,
        "textAlign": "center",
      }
    }
    testID="number-input"
    value="10"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="increment-button"
  >
    <Text />
  </View>
</View>
`;
